code/Helper/Image.php

```diff
@@ -31,16 +31,11 @@ public function toString()
             $model->setWatermark($this->getWatermark());
         }
 
-        return $this->getUrl($model->saveFile());
+        return $this->removeProtocol($model->saveFile()->getUrl());
     }
 
-    public function getUrl(Mage_Catalog_Model_Product_Image $model)
+    public function removeProtocol($url)
     {
-        $baseDir = Mage::getBaseDir();
-        $path = str_replace($baseDir.DS, '', $model->getNewFile());
-
-        $url = str_replace(DS, '/', $path);
-
-        return $url;
+        return str_replace(['https://', 'http://'], '//', $url);
```

Comment:
Yes, we had the same thing in version 1.5.5. But it ended up in issues with placeholder images.
Indexing whole URL is the most straight forward and the most understandable approach as the logic is only at one place and not spread between indexing and front-end part.